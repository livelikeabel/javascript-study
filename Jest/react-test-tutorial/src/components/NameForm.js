import React, {Component} from 'react'

class NameForm extends Component {
    static defaultProps = {
        onInsert: () => console.warn('onInsert not defined')
    }
    state = {
        name: ''
    }
    onChange = (e) => {
        this.setState({name: e.target.value})
    }
    onSubmit = (e) => {
        this.props.onInsert(this.state.name);
        this.setState({name:''})
        e.preventDefault()
    }

    render() {
        return (
            <form onSubmit={this.onSubmit}>
                <label>이름</label>
                <input type="text" value={this.state.name} onChange={this.onChange} />
                <button type="submit">등록</button>
            </form>
        )
    }
}

export default NameForm