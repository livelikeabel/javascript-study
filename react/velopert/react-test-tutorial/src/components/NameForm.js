import React, { Component } from 'react'

class NameForm extends Component {
  static defaultProps = {
    onSubmit: () => console.warn('onSubmit not defined')
  }
  state = {
    name: ''
  }
  onChange = e => {
    this.setState({ name: e.target.value })
  }
  onSubmit = e => {
    const { name } = this.state
    const { onInsert } = this.props
    // 이름을 추가하고, name값 초기화
    onInsert(name)
    this.setState({ name: '' })
    debugger
    e.preventDefault()
  }
  render() {
    const { onSubmit, onChange } = this
    const { name } = this

    return (
      <form onSubmit={onSubmit}>
      <label>NAME : </label>
        <input type="text" onChange={onChange} value={name} />
        <button type="submit">submit</button>
      </form>
    )
  }
}

export default NameForm
