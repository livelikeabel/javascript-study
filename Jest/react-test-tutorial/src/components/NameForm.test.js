import React from 'react'
// import renderer from 'react-test-renderer'
import {shallow} from 'enzyme'
import NameForm from './NameForm'

// describe('NameForm', () => {
//     let component = null;

//     it('renders correctly', () => {
//         component = renderer.create(<NameForm/>)
//     })

//     it('matches snapshot', () => {
//         const tree = component.toJSON();
//         expect(tree).toMatchSnapshot();
//     })
// })

describe('NameForm', () => {
    let component = null;

    it('renders correctly', () => {
        component = shallow(<NameForm />)
    })

    it('matches snapshot', () => {
        expect(component).toMatchSnapshot();
    })
})