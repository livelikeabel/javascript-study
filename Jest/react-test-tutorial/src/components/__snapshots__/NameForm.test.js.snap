// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameForm matches snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <label>
    이름
  </label>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    type="submit"
  >
    등록
  </button>
</form>
`;
